### PR TITLE
feat(frontend): add prerequisite feats for fdw

### PR DESCRIPTION
**Pull request by KeXiangWang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Three prerequisite features for fdw.
1. allow isolation level, but proceed ignoring it.
2. `attcollation` in `pg_catalog.pg_attribute`
3. allow `Read-write` transaction but proceed as `Read-only` transaction.
1 & 3 are not really implemented here. They are allowed to support Risingwave as pg fdw. The corresponding notices are thrown to notify users.

example:
```
START TRANSACTION ISOLATION LEVEL REPEATABLE READ;
COMMIT;
---------------
NOTICE:  Transaction with given Isolation Level is not supported yet.
For compatibility, this statement will still proceed with RepeatableRead.
NOTICE:  Read-write transaction is not supported yet. Please specify `READ ONLY` to start a read-only transaction.
For compatibility, this statement will still succeed and be executed as Read-only transactions.
The write operations in this transaction will be rejected.
START_TRANSACTION
COMMIT
```

## Checklist

- [ ] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by fuyufjh — APPROVED** [review]
The 3 changes LGTM.

File: src/frontend/src/handler/transaction.rs [hunk 1]
```diff
@@ -42,29 +43,42 @@ pub async fn handle_begin(
...
             Some(TransactionAccessMode::ReadOnly) => AccessMode::ReadOnly,
             Some(TransactionAccessMode::ReadWrite) | None => {
                 // Note: This is for compatibility with some external drivers (like psycopg2) that
-                // issue `BEGIN` implicitly for users. Not actually starting a transaction is okay
-                // since `COMMIT` and `ROLLBACK` are no-ops (except for warnings) when there is no
-                // active transaction.
+                // issue `BEGIN` implicitly for users.
                 const MESSAGE: &str = "\
                     Read-write transaction is not supported yet. Please specify `READ ONLY` to start a read-only transaction.\n\
-                    For compatibility, this statement will still succeed but no transaction is actually started.";
```
Comment (BugenZhao) [inline src/frontend/src/handler/transaction.rs:58]: This is interesting. Does the original compatibility manner (i.e., not actually starting a txn) work for `fdw`?

**Review by BugenZhao — COMMENTED** [review]
Rest LGTM

File: src/frontend/src/handler/transaction.rs (hunk 1 above)
Comment (KeXiangWang) [reply src/frontend/src/handler/transaction.rs:58]: For fdw, it should work. But cursors should only work in transaction. If we keep the original compatibility manner we must allow declare/fetch/close cursors out of transaction blocks. This is unexpected and risky. So I am inclined to change the compatibility manner.

**Pull request merged**